light_node/src/configuration.rs

```
@@ -1259,4 +1188,63 @@ impl Environment {
+        // filter all configurations and split to valid and ok
+        let (oks, errors): (Vec<_>, Vec<_>) =
+            self.storage
+                .context_action_recorders
+                .iter()
+                .map(|backend| match backend {
+                    storage::context::actions::ContextActionStoreBackend::RocksDB => {
+                        match storage.merkle_context_actions() {
+                            Some(merkle_context_actions) => Ok(Some(Box::new(
+                                ContextActionStorage::new(merkle_context_actions, storage.seq()),
+                            )
+                                as Box<dyn ActionRecorder + Send>)),
+                            None => Err(InvalidRecorderConfigurationError(
+                                "Missing RocksDB source 'storage.merkle_context_actions()'"
+                                    .to_string(),
+                            )),
+                        }
+                    }
+                    storage::context::actions::ContextActionStoreBackend::FileStorage { path } => {
+                        Ok(Some(Box::new(ActionFileStorage::new(path.to_path_buf()))
+                            as Box<dyn ActionRecorder + Send>))
+                    }
+                    storage::context::actions::ContextActionStoreBackend::NoneBackend => Ok(None),
+                })
+                .partition(Result::is_ok);
+
+        // collect all invalid
+        if !errors.is_empty() {
+            let errors = errors
+                .iter()
+                .filter_map(|e| match e {
+                    Ok(_) => None,
+                    Err(e) => Some(format!("{:?}", e)),
+                })
+                .collect::<Vec<_>>()
+                .join(", ");
+            return Err(InvalidRecorderConfigurationError(format!(
+                "errors: {:?}",
+                errors
+            )));
+        }
+
+        // return just oks
```

Are errors silently ignored? no logging?

---

Oh duh, missed the `return` above.